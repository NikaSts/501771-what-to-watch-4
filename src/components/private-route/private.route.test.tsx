import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {Router} from 'react-router-dom';
import history from '../../history';
import PrivateRoute from './private-route';

const mockStore = configureStore([]);

const authorizationStatus = `NO_AUTH`;

it(`PrivateRoute should render LoginPage if user not authtorized`, () => {
  const store = mockStore({
    USER: {
      authorizationStatus,
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <PrivateRoute
              authorizationStatus={authorizationStatus}
            />
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        })
      .toJSON();

  expect(tree).toMatchSnapshot();
});
