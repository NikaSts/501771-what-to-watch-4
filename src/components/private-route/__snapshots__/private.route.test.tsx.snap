// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute should render LoginPage if user not authtorized 1`] = `null`;
